### ISSUE TITLE
return-generation-logits bug when fp8 enabled

### ISSUE THREAD

**Issue by binhtranmcs** [NVIDIA/TensorRT-LLM#2088]
I am running llama3 model on an rtx4090 with fp8 quantization. In the [result](https://github.com/NVIDIA/TensorRT-LLM/blob/main/cpp/include/tensorrt_llm/executor/executor.h#L323), `outputTokenIds` seems to be correct but the `generationLogits` is all wrong. I also tested the same model without quantization and the returned logits are all correct, so I guess there is something wrong when returning the logits with fp8 enable.

How I tested: I deployed the model using tritonserver with tensorrtllm_backend. I changed the bls backend a bit to get the softmax of the `generationLogits` as well as the tokens generated. I made a call using [client.txt](https://github.com/user-attachments/files/16504748/client.txt) and got the result in [log.txt](https://github.com/user-attachments/files/16504755/log.txt).

Command to run the client: `python3 client.py -p "hello how are you" --model-name tensorrt_llm_bls --request-id testid --verbose -o 10 --return-generation-logits`.

Please have a look. Thanks in advance!

**Comment by binhtranmcs** [comment]
Any update on this? Another insight is that when running with `examples/run.py`, the output generation logits are correct.
Command to run:
```
python3 examples/run.py \
  --engine_dir ENGINE_DIR \
  --tokenizer_dir TOKENIZER_DIR \
  --max_output_len 10 \
  --output_logits_npy \
  --input_text "hello how are you"
```

**Comment by yweng0828** [comment]
Hi, @binhtranmcs , thanks for your feedback.
I have two questions to confirm with you:
1) You mentioned that you made some code changes. Have you verified the logits without making any changes?
2) Could you provide a complete reproduction step (such as convert checkpoint/build engine/run scripts)? So that we can follow up better.

Thanks
Yue

**Comment by binhtranmcs** [comment]
@yweng0828 You can check [here](https://github.com/triton-inference-server/tensorrtllm_backend/issues/598). The problem is that the backend assumes `logits_dtype` to be `float32`, while I can build the engine with `--logits_dtype=float16`. Internally, I updated the backend to only handle `--logits_dtype=float16` by updating the type [here](https://github.com/triton-inference-server/tensorrtllm_backend/blob/main/inflight_batcher_llm/src/model_instance_state.cc#L946) to `int16_t`, as it uses less memory. But I think you may need to update the backend for more general use.
